### PR TITLE
Tweak the suggested warning message for partition procedure

### PR DESCRIPTION
**Pull request by jxchinaren**

### CODE REVIEW

File: tests/frontend/org/voltdb/compiler/TestVoltCompiler.java [hunk 1]
```diff
@@ -380,6 +380,47 @@ public void testMismatchedPartitionParams() throws IOException {
...
+            "</schemas>" +
+            "</database>" +
+            "</project>";
+
+        final File projectFile = VoltProjectBuilder.writeStringToTempFile(simpleProject);
+        final String projectPath = projectFile.getPath();
+
+        final VoltCompiler compiler = new VoltCompiler();
+
+        final boolean success = compiler.compileWithProjectXML(projectPath, testout_jar);
```
Comment (paulmartel) [inline tests/frontend/org/voltdb/compiler/TestVoltCompiler.java]: There must be a better way to set up a simple test than to use the obsolete project.xml.

File: src/frontend/org/voltdb/compiler/ProcedureCompiler.java [hunk 2]
```diff
@@ -757,10 +765,11 @@ static void compileSingleStmtProcedure(VoltCompiler compiler,
                         } else {
                             valueDescription = partitionValue.toString(); // A simple constant value COULD have been a parameter.
                         }
-                        msg = "This procedure would benefit from adding a parameter to be passed the value " + valueDescription +
-                                " and setting the attribute 'partitioninfo=" + partitioning.getFullColumnName() +
-                                ":" + paramCount  + "'";
+                        msg += "adding a parameter to be passed the value " + valueDescription + " and ";
                     }
+                    msg += "adding 'PARTITION PROCEDURE "+ shortName + " ON TABLE " + tableName + " COLUMN " +
```
Comment (paulmartel) [inline src/frontend/org/voltdb/compiler/ProcedureCompiler.java]: "PARTITION PROCEDURE" is ALSO old syntax -- I don't want to have to change this message again in for VoltDB 6.x.

File: src/frontend/org/voltdb/compiler/ProcedureCompiler.java [hunk 3]
```diff
@@ -757,10 +765,11 @@ static void compileSingleStmtProcedure(VoltCompiler compiler,
                         } else {
                             valueDescription = partitionValue.toString(); // A simple constant value COULD have been a parameter.
                         }
-                        msg = "This procedure would benefit from adding a parameter to be passed the value " + valueDescription +
-                                " and setting the attribute 'partitioninfo=" + partitioning.getFullColumnName() +
-                                ":" + paramCount  + "'";
+                        msg += "adding a parameter to be passed the value " + valueDescription + " and ";
```
Comment (paulmartel) [inline src/frontend/org/voltdb/compiler/ProcedureCompiler.java]: This edge case where we need to advise "adding a parameter to be passed the value " should have a test case.
The simplest example has query text something like "WHERE PARTITION_KEY = 1".

Comment (jxchinaren) [inline src/frontend/org/voltdb/compiler/ProcedureCompiler.java]: Yes, it's there at: 
tests/frontend/org/voltdb/compiler/TestVoltCompilerAnnotationsAndWarnings.java:181
